src/toolbar/sticky/stickytoolbarview.js

```diff
@@ -20,21 +20,88 @@ export default class StickyToolbarView extends ToolbarView {
 	/**
 	 * @inheritDoc
 	 */
-	constructor() {
-		super();
+	constructor( locale, limiterElement ) {
+		super( locale );
 
 		const bind = this.bind;
 
 		this.model.set( 'isSticky', false );
+		this.model.set( 'isStickyToLimiterBottom', false );
+		this.model.set( 'left', null );
+		this.model.set( 'marginLeft', null );
+
+		/**
+		 * The limiter element for the sticky toolbar instance. Its bounding rect limits
+		 * the "stickyness" of the toolbar, i.e. when the toolbar reaches the bottom
+		 * edge of the limiter, it becomes sticky to that edge and does not float
+		 * off the limiter.
+		 *
+		 * @member {HTMLElement} ui.stickyToolbar.StickyToolbarView#limiterElement
+		 */
+		this.limiterElement = limiterElement;
 
 		Template.extend( this.template, {
 			attributes: {
-				// Toggle class of the toolbar when "sticky" state changes in the model.
-				class: bind.if( 'isSticky', 'ck-toolbar_sticky' )
+				class: [
+					// Toggle class of the toolbar when "sticky" state changes in the model.
+					bind.if( 'isSticky', 'ck-toolbar_sticky' ),
+					bind.if( 'isStickyToLimiterBottom', 'ck-toolbar_sticky_bottom-limit' ),
+				],
+				style: {
+					width: bind.to( 'isSticky', ( isSticky ) => {
+						// 2px compensates the border.
+						return isSticky ? pixelize( this._elementPlaceholder.getBoundingClientRect().width + 2 ) : null;
+					} ),
+
+					top: bind.to( 'isStickyToLimiterBottom', ( isStickyToLimiterBottom ) => {
+						return isStickyToLimiterBottom ?
+							pixelize( window.scrollY + this._limiterRect.bottom - this._toolbarRect.height ) : null;
+					} ),
+
+					left: bind.to( 'left' ),
+					marginLeft: bind.to( 'marginLeft' )
+				}
 			}
 		} );
 
 		/**
+		 * A dummy element which visually fills the space as long as the
+		 * actual toolbar is sticky. It prevents flickering of the UI.
+		 *
+		 * @private
+		 * @property {HTMLElement} ui.stickyToobar.StickyToolbarView#_elementPlaceholder
+		 */
+		this._elementPlaceholder = new Template( {
+			tag: 'div',
+			attributes: {
+				class: [
```

Comment:
I'll bother you about this every time I see this – there should be an option to define classes and styles next to the `tag`. Creating a whole object just set the two most popular propos looks bad.